### DISCUSSION TITLE
How likely is database corruption?

### DISCUSSION THREAD

**Discussion by macruspareto** [keepassxreboot/keepassxc#6539] [Q&A]
I used to store my passwords in txt files in Truecrypt (then Veracrypt) container.
For 10+ years I had accidental reboots, BSoDs, dirty unmounts, power loss but the container has never got rendered inaccessible/damaged. Very fail-safe.

How fault-tolerant is the KeePass design?
After every change, I backup my .kdbx in three separate locations (USB and NAS), with older copies aswell so I could rollback.

But still, in your opinion, is there a risk of database corruption? Mainly due to file I/O, overwriting and re-encrypting, improper unmounting etc?

**Comment by droidmonkey** [comment]
I've been using keepassxc since 2016, keepass2 before that. Since developing keepassxc I've eaten my own dog food and run on develop snapshots. I store my db in onedrive and use keepassdx on mobile. I have never once had a corrupt or inaccessible database. I've never corrupted a test database either during strange debugging scenarios. Your likliest cause of corruption is usb drives that are not properly unmounted.

**Reply by michaelk83** [reply]
Nevertheless, there has been some complaints from users experiencing some issues. But as noted, the most likely culprit is hardware problems. There is also a known difficulty with network drives of various kinds, which is why there are 3 different save methods, trying to work around that.

My recommendations are:
- Prefer to load and save locally, to an internal drive.
- If needed, *copy* (not move!) to/from the network location separately, once the file is inert.
- Keep at least 1-2 backups on *reliable* storage hardware.

I personally have used local and Google Drive backups, and have not had corruption problems.

(edit: didn't notice how old this question was)

**Comment by jasperf** [comment]
Had multiple warnings this morning my database stored on Google Drive might be corrupted or inaccessible. Then I restarted application and opened laptop to allow for finger print unlocking and all was well again. Got me worried there for some though.
